include/pybind11/pybind11.h

```diff
@@ -2854,6 +2855,51 @@ function get_override(const T *this_ptr, const char *name) {
     PYBIND11_OVERRIDE_PURE_NAME(                                                                  \
         PYBIND11_TYPE(ret_type), PYBIND11_TYPE(cname), #fn, fn, __VA_ARGS__)
 
+/** \rst
+    Macro to wrap :c:macro:`PYBIND11_OVERRIDE_PURE_NAME` and :c:macro:`PYBIND11_OVERRIDE_PURE`
+    depending on the base class and cname parameter provided.
+    See :ref:`_multiple_inheritance` for more information.
+
+    .. code-block:: cpp
+
+      template<class AnimalBase = Animal>
+      class PyAnimal : public AnimalBase {
+      public:
+          // Inherit the constructors
+          using AnimalBase::AnimalBase;
+
+          // Trampoline (need one for each virtual function)
+          std::string go(int n_times) override {
+              PYBIND11_OVERRIDE_TEMPLATE(
+                  Animal,      // The base class containing the purely virtual implementation
+                  std::string, // Return type (ret_type)
+                  Dog,         // Parent class (cname)
+                  "_go",       // Name of method in Python (name)
+                  go,          // Name of function in C++ (must match Python name) (fn)
+                  n_times      // Argument(s) (...)
+              );
+          }
+      };
+\endrst */
+#define PYBIND11_OVERRIDE_TEMPLATE_NAME(base, ret_type, cname, name, fn, ...)                     \
+    if (std::is_same<base, cname>::value) {                                                       \
+        PYBIND11_OVERRIDE_PURE_NAME(                                                              \
+            PYBIND11_TYPE(ret_type), PYBIND11_TYPE(cname), name, fn, __VA_ARGS__);                \
+    } else {                                                                                      \
+        PYBIND11_OVERRIDE_NAME(                                                                   \
+            PYBIND11_TYPE(ret_type), PYBIND11_TYPE(cname), name, fn, __VA_ARGS__);                \
+    }
```

Comment:
This isn't a lot of meat, especially given the total size of this PR.

There is no significant complexity in this macro, and it's catering only to a niche case.

In contrast, the added documentation is overwhelming. If that's reflective of a typical use case, having to implement this helper macro is a relatively insignificant amount of extra code.

I don't see a lot of value added this pair of macros (this one and the one below).

Reply:
Thank you Ralf for having a look at my PR. I understand your sentiment and it is perfectly fine to not merge this. The only thing I would like to point out is that the first part of the PR is a pure documentation update. It shows a trampoline class pattern for combining two (or more) separate class inheritance
hierarchies avoiding code duplication (backed by some unit tests). Only the second part requires the new macros for reducing code duplication with pure virtual functions. Cheers Arnim

Reply:
I believe
* it's too much detail and too much code in that part of the documentation. (Especially because we're not set up to unit-test code in documentation.)
* that view is biased by: I wouldn't want to encourage creating that kind of complexity. I live more by "minimize use of multiple inheritance", it tends to get too confusing too quickly. (Which absolutely does **not** mean "never" btw.)

@henryiii do you think the code in the documentation under this PR could be a useful addition as an example somewhere?